packages/get-platform/src/getPlatform.ts

```diff
@@ -255,7 +264,33 @@ export async function getSSLVersion(args: GetOpenSSLVersionParams): Promise<GetO
 }
 
 export async function getPlatform(): Promise<Platform> {
-  const { platform, libssl, distro, arch } = await getos()
+  const { platform, distro, arch, libssl } = await getos()
+
+  // sometimes we fail to detect the libssl version to use, so we default to 1.1.x
+  const defaultLibssl = '1.1.x' as const
+  if (platform === 'linux' && libssl === undefined) {
+    /**
+     * Ask the user to install openssl manually, and provide some additional instructions based on the detected Linux distro.
+     */
+    const additionalMessage = match({ distro })
+      .with({ distro: 'debian' }, () => {
+        return "Please manually install openssl via `apt-get update -y && apt-get install -y openssl` and try installing Prisma again. If you're running Prisma on Docker, you may also try to replace your base image with `node:lts-slim`, which already ships with openssl installed."
+      })
+      .otherwise(() => {
+        return 'Please manually install openssl and try installing Prisma again.'
+      })
+
+    console.warn(`Prisma failed to detect the libssl/openssl version to use, and may not work as expected. Defaulting to "openssl-${defaultLibssl}".
+${additionalMessage}`)
+  }
+
+  // sometimes we fail to detect the distro in use, so we default to debian
+  const defaultDistro = 'debian' as const
+  if (platform === 'linux' && distro === undefined) {
+    console.warn(
+      `Prisma failed to detect the Linux distro in use, and may not work as expected. Defaulting to "${defaultDistro}".`,
```

Comment:
I think this is a very nice warning, but I can also see how annoying it would be to me if I still used Arch for my Linux system 😄

Can `getPlatform` be called multiple times during the lifetime of a Prisma client? Could we make sure these warnings are only printed once? We have a `warnOnce` utility for that in `@prisma/internals` which you could use or re-implement.

It would generally be preferable to use our existing warnings infrastructure in `@prisma/internals` as that would avoid code duplication, provide consistent styling and respect the `PRISMA_DISABLE_WARNING` env var, but I'm not sure without checking if we can do that without circular dependencies between packages.

Reply:
`getPlatform` is already memoized in `BinaryEngine.ts`. In `LibraryEngine.ts`, it's not, but it's only called once in case of errors

Reply:
> getPlatform is already memoized in BinaryEngine.ts. In LibraryEngine.ts, it's not, but it's only called once in case of errors

I'd still not rely on this fact as it could easily be changed accidentally.

Reply:
I would treat this problem separately only if users complain about too many warnings. Our focus is on popular systems also used on Docker, like Alpine and Debian

Reply:
I am fine with potentially shipping to many warnings for now.

Reply:
I'm totally fine with doing it in a follow-up PR, but I'd still prefer to have the consistent warnings with styling and functionality that the logger from `@prisma/internals` provides if it's possible and doesn't introduce circular dependencies.

Reply:
I've copied the functionality needed from `@prisma/internals` - unfortunately it was either that or circular dependencies.